src/Streamly/Internal/Data/Parser/ParserD/Types.hs

```diff
@@ -426,7 +426,11 @@ alt (Parser stepL initialL extractL) (Parser stepR initialR extractR) =
             Error err -> Error err
 
     extract (AltParseR sR) = extractR sR
-    extract (AltParseL _ sL) = extractL sL
+    extract (AltParseL _ sL) = do
+        r <- try $ extractL sL
+        case r of
+            Left (_ :: ParseError) -> initialR >>= extractR
```

Comment:
This will just create more confusion in my opinion. Until we find a proper solution, we might as well use `extractL sL`  and note that `alt` does not work for parsers that fail without sufficient input. Or as @pranaysashank suggests, remove the `alt` instance for now.